### PR TITLE
Improve Hotkeys and Enhance Tabs

### PR DESCRIPTION
**Pull request by ron-murhammer**
Addresses: 
https://forums.triplea-game.org/topic/690/improve-hotkeys
https://forums.triplea-game.org/topic/702/improve-resources-tab
https://forums.triplea-game.org/topic/678/naval-bases-attribute

**Functional Changes**
1. Remove WASD as this already conflicts with Ctrl+S and I don't think is used much
2. Add Ctrl+hotkey for each tab panel
-- Ctrl+A - Actions
-- Ctrl+P - Stats (rename to Players)
-- Ctrl+R - Economy (rename to Resources)
-- Ctrl+O - Objectives
-- Ctrl+N - Notes
-- Ctrl+T - Territory
3. Add Ctrl+(-/+) to zoom out/in
4. Ctrl+E - Toggle edit mode
5. Ctrl+H & Ctrl+G - Toggle history and game mode
6. Ctrl+W - Toggle show politics panel (W for war)
7. Ctrl+L - Toggle unit help list
8. Ctrl+F - Toggle full screen (hide all tabs and minimap)
9. Add naval and air base info to territory tab
10. Add income and icons to resources tab:
image

### CODE REVIEW

**Comment by DanVanAtta** [comment]
Good set of changes included :+1: 

One item I'd like to discuss:
> Ctrl+F - Toggle full screen (hide all tabs and minimap)

In the cat-walked-across keyboard scenario, or just mistyping keys and accidently doing a 'ctrll+f', how easy will it be to 'undo' or even know you're in full-screen mode? I think ideally we'd have a small game icon somewhere near the minimap to turn full screen on and off to go with the hotkey.

Some options:
- fast follow to add an indicator to let user know when they're in full screen mode
- Tackle the indicator in this PR
- Hold off on ctrl+f until there is a matching UI element to indicate full screen mode and another to turn it on and off

**Comment by ron-murhammer** [comment]
@DanVanAtta There is technically an indicator already. As you can click the small arrows (><), to expand/minimize the right panel.

**Comment by DanVanAtta** [comment]
@ron-murhammer moot concern for cat-walking on keyboard then :smile_cat: ;
 Can we attach a hover-text to the indicator?

**Comment by DanVanAtta** [comment]
Can we, and would it be any trouble to add hover-texts on the panel tabs too to tell people the hotkey when hovering? The idea is that it would give users a way to discover the hotkeys in-context without digging through the shortcut help menu.

File: game-core/src/main/java/games/strategy/triplea/ui/AbstractStatPanel.java [hunk 1]
```diff
@@ -39,13 +41,17 @@
    * @return all the alliances with more than one player.
    */
   public Collection<String> getAlliances() {
-    final Collection<String> alliances = new TreeSet<>();
+    return getAllianceMap().keySet();
+  }
+
+  public Map<String, Set<PlayerID>> getAllianceMap() {
```
Comment (DanVanAtta) [inline game-core/src/main/java/games/strategy/triplea/ui/AbstractStatPanel.java]: can this be protected? 

Also have you considered using guava multimap? I found they are funky to construct, but beyond that it is a data structure that is exactly meant for `Map<T, Collection<T>>`: http://www.baeldung.com/guava-multimap

File: game-core/src/main/java/games/strategy/triplea/ui/EconomyPanel.java [hunk 2]
```diff
@@ -76,22 +93,40 @@ private synchronized void loadData() {
...
-          for (int i = 0; i < statsResource.length; i++) {
-            collectedData[row][i + 1] =
-                statsResource[i].getFormatter().format(statsResource[i].getValue(player, gameData));
+          final IntegerMap<Resource> resourceIncomes = AbstractEndTurnDelegate.findEstimatedIncome(player, gameData);
+          resourceIncomeMap.put(player, resourceIncomes);
+          for (int i = 0; i < resourceStats.size(); i++) {
+            final ResourceStat resourceStat = resourceStats.get(i);
+            final Resource resource = resourceStat.resource;
+            final double quantity = resourceStat.getValue(player, gameData);
+            final StringBuilder text = new StringBuilder(resourceStat.getFormatter().format(quantity) + " (");
```
Comment (DanVanAtta) [inline game-core/src/main/java/games/strategy/triplea/ui/EconomyPanel.java]: `text` is a bit vague, when we get to the line: ` collectedData[row][i + 1] = text.toString();`; it's a bit hard to understand as the terminology is overly generic. Something like `playerResourceIncome` could work, I am having trouble figuring out exactly which text is going into this variable.

For this code block extracting to a well named private static helper method could help. The append with a potential `()` or `(+)` is hard to understand; I'll need to study this further, but I'm curious if that really is the result output and where that is used.

File: game-core/src/main/java/games/strategy/triplea/ui/EconomyPanel.java [hunk 3]
```diff
@@ -76,22 +93,40 @@ private synchronized void loadData() {
...
-                statsResource[i].getFormatter().format(statsResource[i].getValue(alliance, gameData));
+        for (final Entry<String, Set<PlayerID>> alliance : allianceMap.entrySet()) {
+          collectedData[row][0] = alliance.getKey();
+          for (int i = 0; i < resourceStats.size(); i++) {
+            final ResourceStat resourceStat = resourceStats.get(i);
+            final double quantity = resourceStat.getValue(alliance.getKey(), gameData);
+            final StringBuilder text = new StringBuilder(resourceStat.getFormatter().format(quantity) + " (");
+            final int income = alliance.getValue().stream()
+                .mapToInt(p -> resourceIncomeMap.get(p).getInt(resourceStat.resource)).sum();
+            if (income >= 0) {
```
Comment (DanVanAtta) [inline game-core/src/main/java/games/strategy/triplea/ui/EconomyPanel.java]: deja-vu

Please do extract to a function,most of the bodies of these for loops look about the same, can be pretty cleanly parameterized with resourceStat and the conditional check value (eg: `resourceIncomes.getInt(resource) >= 0`)

File: game-core/src/main/java/games/strategy/triplea/ui/AbstractStatPanel.java (hunk 1 above)
Comment (ron-murhammer) [reply game-core/src/main/java/games/strategy/triplea/ui/AbstractStatPanel.java]: Updated visibility. Not really that big of a fan of multimap.

File: game-core/src/main/java/games/strategy/triplea/ui/EconomyPanel.java (hunk 2 above)
Comment (ron-murhammer) [reply game-core/src/main/java/games/strategy/triplea/ui/EconomyPanel.java]: Extracted helper method.

File: game-core/src/main/java/games/strategy/triplea/ui/EconomyPanel.java (hunk 3 above)
Comment (ron-murhammer) [reply game-core/src/main/java/games/strategy/triplea/ui/EconomyPanel.java]: Extracted helper method.

**Comment by ron-murhammer** [comment]
@DanVanAtta Points should be addressed. Only thing that can't easily be done is a tooltip on the JSplitPane for fullscreen as the tooltip doesn't display when hovering over it.

**Comment by DanVanAtta** [comment]
Thanks! good work

**Pull request merged**